### PR TITLE
[K32W] Improve debugging + CPU speed

### PR DESCRIPTION
**Pull request by doru91**
.#### Problem
* Improve logging capabilities
* Increase CPU speed

#### Change overview
Add a timestamp for platform logging. Also increase the CPU speed
from 32Mhz to 48Mhz

#### Testing
* manually tested

### CODE REVIEW

**Review by andy31415 — APPROVED** [review]

**Review by woody-apple — APPROVED** [review]

File: src/platform/K32W/Logging.cpp [hunk 1]
```diff
@@ -21,26 +21,29 @@
...
 {
+    sprintf(buf, "[%lu]", otPlatAlarmMilliGetNow());
+
     if (chipCategory != kLogCategory_None)
     {
         switch (chipCategory)
         {
         case kLogCategory_Error:
-            memcpy(buf, "[Error]", 7);
+            memcpy(buf + strlen(buf), "[Error]", 7);
```
Comment (msandstedt) [inline src/platform/K32W/Logging.cpp:39]: I don't understand how this was supposed to work before or how it works now.  buf isn't length delimited, but no null terminator is explicitly written.

Why not just use strcpy?  That would give you the null character and also alleviate the need to count chars.

Comment (jelderton) [reply src/platform/K32W/Logging.cpp:39]: Seems like a strcat() scenario.

File: src/platform/K32W/Logging.cpp [hunk 2]
```diff
@@ -21,26 +21,29 @@
 static bool isLogInitialized;
 extern uint8_t gOtLogUartInstance;
 extern "C" void K32WWriteBlocking(const uint8_t * aBuf, uint32_t len);
+extern "C" uint32_t otPlatAlarmMilliGetNow(void);
 
 namespace chip {
 namespace Logging {
 namespace Platform {
 
 void GetMessageString(char * buf, uint8_t chipCategory, uint8_t otLevelLog)
```
Comment (mspang) [inline src/platform/K32W/Logging.cpp:30]: Can you add a buffer length parameter please?

**Review by mspang — CHANGES_REQUESTED** [review]

File: src/platform/K32W/Logging.cpp (hunk 1 above)
Comment (doru91) [reply src/platform/K32W/Logging.cpp]: @msandstedt  @jelderton , agree that the old code was messy. I refactored it using the snprintf function.

File: src/platform/K32W/Logging.cpp (hunk 2 above)
Comment (doru91) [reply src/platform/K32W/Logging.cpp]: @mspang done. Thanks!

**Review by msandstedt — APPROVED** [review]

File: src/platform/K32W/Logging.cpp [hunk 3]
```diff
@@ -21,60 +21,54 @@
...
 
-void GetMessageString(char * buf, uint8_t chipCategory, uint8_t otLevelLog)
+void GetMessageString(char * buf, uint8_t bufLen, const char * module, uint8_t category)
 {
-    if (chipCategory != kLogCategory_None)
+    int writtenLen = 0;
+    const char * categoryString;
+
+    writtenLen = snprintf(buf, bufLen, "[%lu]", otPlatAlarmMilliGetNow());
+    assert((writtenLen > 0) && (writtenLen < bufLen));
```
Comment (bzbarsky-apple) [inline src/platform/K32W/Logging.cpp:36]: Why is this assert valid?  What guarantees that `writtenLen >= bufLen` can't happen here?

File: src/platform/K32W/Logging.cpp [hunk 4]
```diff
@@ -21,60 +21,54 @@
...
-            memcpy(buf, "[Info]", 6);
-            break;
-        case OT_LOG_LEVEL_DEBG:
-            memcpy(buf, "[Debug]", 7);
-            break;
+            categoryString = "U";
         }
+
+        writtenLen = snprintf(buf + writtenLen, bufLen, "[%s]", categoryString);
+        assert((writtenLen > 0) && (writtenLen < bufLen));
```
Comment (bzbarsky-apple) [inline src/platform/K32W/Logging.cpp:57]: Again, what guaratees the assert holds?  I don't see anything.

File: src/platform/K32W/Logging.cpp [hunk 5]
```diff
@@ -21,60 +21,54 @@
...
+        bufLen -= writtenLen;
     }
+
+    writtenLen = snprintf(buf + writtenLen, bufLen, "[%s]", module);
+    assert((writtenLen > 0) && (writtenLen < bufLen));
```
Comment (bzbarsky-apple) [inline src/platform/K32W/Logging.cpp:62]: And here.

**Review by bzbarsky-apple — CHANGES_REQUESTED** [review]

File: src/platform/K32W/Logging.cpp (hunk 3 above)
Comment (mspang) [reply src/platform/K32W/Logging.cpp:36]: > Why is this assert valid? What guarantees that `writtenLen >= bufLen` can't happen here?

We can calculate the minimum size and document this as a precondition?

**Pull request closed without merging**